compiler-core/src/language_server/annotation_printer.rs

```diff
@@ -0,0 +1,218 @@
+#![allow(dead_code)]
+
+use ecow::EcoString;
+use im::HashMap;
+use itertools::Itertools;
+use std::sync::Arc;
+
+use crate::{
+    ast::TypeAst,
+    type_::{Type, TypeVar, PRELUDE_MODULE_NAME},
+};
+
+#[derive(Debug)]
+pub struct AnnotationPrinter<'a> {
+    names: HashMap<u64, EcoString>,
+    uid: u64,
+
+    // A mapping of printd type names to the module that they are defined in.
+    printed_types: HashMap<EcoString, EcoString>,
+
+    // Mapping of module.type to type alias -
+    // eg: import mod1.{type Cat as c} -> Key: (mod1, Cat), Value: c
+    // eg: type UserId = Int -> Key: (mod, UserId), Value: Int
+    type_aliases: &'a HashMap<(EcoString, EcoString), EcoString>,
+
+    // Mapping of module alias to module name
+    module_aliases: &'a HashMap<EcoString, EcoString>,
+    type_parameters: &'a HashMap<u64, TypeAst>,
+    current_module: EcoString,
+}
+
+impl<'a> AnnotationPrinter<'a> {
+    pub fn new(
+        type_aliases: &'a HashMap<(EcoString, EcoString), EcoString>,
```

Comment:
Would it be better if we make the constructor just take in the module as an argument ?  We could then construct the necessary context in the constructor.

Reply:
Could be good, however I think that would make it a lot harder to write the tests as there would be a great deal more information that would be needed to construct this class.